js/src/util/backdrop.js

```diff
@@ -138,6 +142,12 @@ class Backdrop extends Config {
       execute(this._config.clickCallback)
     })
 
+    if (this._config.keydownCallback) {
+      EventHandler.on(element.parentElement, EVENT_KEYDOWN, event => {
```

Comment:
Can you explain me, why you set the listener over the parent element?

Reply:
> Can you explain me, why you set the listener over the parent element?

The `backdrop`, being a part of another element, is always out of focus (e.g. the focus is on the `modal`/`offcanvas` element itself), and therefore a `keydown` event will not be triggered on it.
I double-checked my code and I agree that using the parent element isn't the best approach, since it doesn't work if the `offcanvas` is placed in another div.
According to the documentation only one `offcanvas`/ `modal` can be opened at once, and these are the only two elements using the `backdrop` so I think a safe solution is setting the event listener on the `document` element.

Reply:
this being said, I am not sure if we need to change something on offcanvas. 

The initial point is to enable `esc` behavior in case of static backdrop. Another approach that seems fine too, is `focusTrap` to trap clicks too, in order not to let user get focus from the trapped element. `Esc` would work in that case too


/cc @patrickhlauke

Reply:
> The initial point is to enable `esc` behavior in case of static backdrop. Another approach that seems fine too, is `focusTrap` to trap clicks too, in order not to let user get focus from the trapped element. `Esc` would work in that case too

I've explored the second approach as well, and it is also a possibility. Personally, I think this should remain a `backdrop` functionality for now, since this discussion is about the wanted behavior of the `backdrop` and not of any element in focus. There might be a place in the future for extending the `focustrap` but I'm not sure it is currently needed.

Reply:
@patrickhlauke @twbs/js-review your opinion would help here